valohai/distributed.py

```diff
@@ -0,0 +1,69 @@
+import json
+from typing import List, Optional
+
+from valohai import paths
+from valohai.internals.distributed_config import DistributedConfig, Member
+from valohai.internals.distributed_config.consts import MASTER_MEMBER_ID
+from valohai.internals.distributed_config.utils import member_id_to_rank
+
+
+class Distributed:
+    """
+    Distributed toolkit accessed through `valohai.distributed`.
+    """
+
+    _config: Optional[DistributedConfig] = None
+
+    def is_distributed_task(self) -> bool:
+        # not a property to mimic `is_running_in_valohai`
+        try:
+            return bool(self.config.group_name)
+        except FileNotFoundError:
+            return False
+        except json.decoder.JSONDecodeError:
+            return False
+
+    @property
+    def group_name(self) -> str:
+        return self.config.group_name
+
+    @property
+    def member_id(self) -> str:
+        return self.config.member_id
+
+    @property
+    def rank(self) -> int:
+        return member_id_to_rank(self.member_id)
+
+    @property
+    def required_count(self) -> int:
+        return self.config.required_count
+
+    def members(self) -> List[Member]:
+        return self.config.members
+
+    def member(self, member_id: str) -> Member:
+        for member in self.members():
+            if member.member_id == member_id:
+                return member
+        raise RuntimeError(f"No member with id {member_id}")
+
+    def self(self) -> Member:
+        return self.config.self
+
+    def master(self) -> Member:
+        return self.member(member_id=MASTER_MEMBER_ID)
+
+    @property
+    def config(self) -> DistributedConfig:
+        if not self._config:
+            with open(self._get_config_path()) as json_file:
+                json_data = json.load(json_file)
+            self._config = DistributedConfig.from_json_data(json_data)
```

Comment:
yup yup, `FileNotFoundError` is already mentioned in the tests and in the `is_distributed_task` function call

the `FileNotFoundError` already reports which file is missing so figured it would enough for starters

I did start to fiddle with adding a separate exception hierarchy starting with `ValohaiError(Exception)` (as that is how I would usually structure library error handling) and figured it would be too much of change to change all exception classes in the project _included_ in a unrelated PR (not much work, but quite a lot of changes happening at once)